python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -223,7 +227,26 @@ def __init__(
             raise ValueError(
                 f"Handoff names must be unique from tool names. Handoff names: {handoff_tool_names}; tool names: {tool_names}"
             )
-        self._model_context: List[LLMMessage] = []
+
+    def remove_all_tools(self) -> None:
+        """Remove all tools."""
+        self._tools = []
+
+    def remove_tools(self, tool_names: List[str]) -> None:
+        """Remove tools by name."""
+        for name in tool_names:
+            for tool in self._tools:
+                if tool.name == name:
+                    self._tools.remove(tool)
+                    break
+            for tool in self._handoff_tools:
```

Comment:
We shouldn't mix handoffs with tools. Let's not change handoffs in this API method.